src/core/attribution/markovChain.test.js

```diff
@@ -174,40 +185,61 @@ describe("core/attribution/markovChain", () => {
         [0.5, 0, 0.25, 0, 0.25],
         [0.5, 0.25, 0, 0.25, 0],
       ]);
-      const pi = await findStationaryDistribution(chain, {
+      const result = await findStationaryDistribution(chain, {
         maxIterations: 255,
         convergenceThreshold: 1e-7,
         verbose: false,
         yieldAfterMs: 1,
       });
-      expectStationary(chain, pi);
+
+      expect(result.convergenceDelta).toBeLessThanOrEqual(1e-7);
+      expect(result.nIterations).toBeLessThanOrEqual(255);
+
+      // This should be wholly deterministic, so having an inline snapshot
+      // should not produce test flakes.
+      // This value should not be updated without a good reason (i.e. we
+      // realized that our convergence semantics were wrong)
+      expect(result.convergenceDelta).toMatchInlineSnapshot(
+        `9.536743161842054e-8`
```

Comment:
Curious on your thought process when writing these snapshot tests. Do
they seem “totally reasonable”, or are they “uncomfortable but better
than nothing”, or something else?

I agree that they’re not _flaky_ in the sense that the code is
deterministic. But they’re certainly brittle: valid mathematical
transformations to the implementation may change the results. For
example, changing `sparseMarkovChainInto` to iterate over the vertices
in reverse order will yield different values:

    > 0.1 + 0.2 + 0.3 == 0.3 + 0.2 + 0.1
    false

(For this particular set of tests, I was careful to find graphs that
were both nontrivial _and_ had transition matrices all of whose nonzero
entries are small integer multiples of reciprocal powers of two, so this
particular problem shouldn’t arise, but I hope that the point stands.)

Suppose that I make a change that causes this test to fail: the expected
convergence delta changes from `9.54e-8` to `9.58e-8` and the number of
iterations increases by 1. A priori, this could be a genuine off-by-one
error, or it could just be due to the fact that floating-point numbers
do not model the theory of reals. How will this test help me determine
whether I’ve actually made a mistake?

Reply:
My rationale for the snapshot tests was that they would act as a sentinel that the behavior had changed at all. But I agree they aren't actually that valuable, so I've removed them.